### PR TITLE
fixed 🐛 Bug Report: defaultCommitMessage #19844

### LINKED ISSUE #19844: 🐛 Bug Report: defaultCommitMessage doesn't work
### 📜 Description

The configuration defaultCommitMessage in app-config.yaml doesn't work. Dependencies have been reviewd and i'ts ok (I checked the README.md https://github.com/backstage/backstage/tree/master/plugins/scaffolder-backend).

appconfig1

And I tested with double quotes like documentation (https://backstage.io/docs/features/software-templates/configuration/)

appconfig2

Integration: GitHub

### 👍 Expected behavior

At my job we need to set jira issue in commit message to push anything in repositories. The configuration is essential to process of POC happens.

### 👎 Actual Behavior with Screenshots

Evidence from the repository created by Backstage

Component created in Backstage:
bs-create-component

Repository:
repository

### 👟 Reproduction steps

Configuration from https://backstage.io/docs/features/software-templates/configuration/

### 📃 Provide the context for the Bug.

_No response_

### 🖥️ Your Environment

yarn run v1.22.19
$ /mnt/d/repositorio/backstage/node_modules/.bin/backstage-cli info
OS:   Linux 5.15.90.1-microsoft-standard-WSL2 - linux/x64
node: v18.16.0
yarn: 1.22.19
cli:  0.22.12 (installed)
backstage:  1.17.0

Dependencies:
  @backstage/app-defaults                          1.4.2
  @backstage/backend-app-api                       0.5.2
  @backstage/backend-common                        0.19.4
  @backstage/backend-dev-utils                     0.1.1
  @backstage/backend-openapi-utils                 0.0.3
  @backstage/backend-plugin-api                    0.6.2
  @backstage/backend-tasks                         0.5.7
  @backstage/catalog-client                        1.4.3
  @backstage/catalog-model                         1.4.1
  @backstage/cli-common                            0.1.12
  @backstage/cli-node                              0.1.3
  @backstage/cli                                   0.22.12
  @backstage/config-loader                         1.4.0
  @backstage/config                                1.0.8
  @backstage/core-app-api                          1.9.1
  @backstage/core-components                       0.13.4
  @backstage/core-plugin-api                       1.5.3
  @backstage/errors                                1.2.1
  @backstage/eslint-plugin                         0.1.3
  @backstage/integration-aws-node                  0.1.5
  @backstage/integration-react                     1.1.18
  @backstage/integration                           1.6.2
  @backstage/plugin-api-docs                       0.9.10
  @backstage/plugin-app-backend                    0.3.50
  @backstage/plugin-app-node                       0.1.2
  @backstage/plugin-auth-backend                   0.18.8
  @backstage/plugin-auth-node                      0.2.19
  @backstage/plugin-catalog-backend                1.12.4
  @backstage/plugin-catalog-common                 1.0.15
  @backstage/plugin-catalog-graph                  0.2.35
  @backstage/plugin-catalog-import                 0.9.13
  @backstage/plugin-catalog-node                   1.4.3
  @backstage/plugin-catalog-react                  1.8.3
  @backstage/plugin-catalog                        1.12.4
  @backstage/plugin-events-node                    0.2.11
  @backstage/plugin-github-actions                 0.6.4
  @backstage/plugin-org                            0.6.13
  @backstage/plugin-permission-common              0.7.7
  @backstage/plugin-permission-node                0.7.13
  @backstage/plugin-permission-react               0.4.14
  @backstage/plugin-proxy-backend                  0.3.2
  @backstage/plugin-scaffolder-backend             1.16.5
  @backstage/plugin-scaffolder-common              1.4.0
  @backstage/plugin-scaffolder-node                0.2.2
  @backstage/plugin-scaffolder-react               1.5.4
  @backstage/plugin-scaffolder                     1.14.4
  @backstage/plugin-search-backend-module-catalog  0.1.6
  @backstage/plugin-search-backend-module-pg       0.5.11
  @backstage/plugin-search-backend-module-techdocs 0.1.6
  @backstage/plugin-search-backend-node            1.2.6
  @backstage/plugin-search-backend                 1.4.2
  @backstage/plugin-search-common                  1.2.5
  @backstage/plugin-search-react                   1.6.4
  @backstage/plugin-search                         1.3.6
  @backstage/plugin-tech-radar                     0.6.7
  @backstage/plugin-techdocs-backend               1.6.7
  @backstage/plugin-techdocs-module-addons-contrib 1.0.18
  @backstage/plugin-techdocs-node                  1.7.6
  @backstage/plugin-techdocs-react                 1.1.9
  @backstage/plugin-techdocs                       1.6.8
  @backstage/plugin-user-settings                  0.7.8
  @backstage/release-manifests                     0.0.9
  @backstage/test-utils                            1.4.2
  @backstage/theme                                 0.4.1
  @backstage/types                                 1.1.0
  @backstage/version-bridge                        1.0.4

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

No, but I'm happy to collaborate on a PR with someone else

### PR DESCRIPTION
**Pull request by npiyush97**
#### :heavy_check_mark: Checklist
closes #19844 

- [X] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [X] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by npiyush97** [comment]
@benjdlambert @camilaibs same with azure,gerrit,gitlab etc ?

File: plugins/scaffolder-backend/src/scaffolder/actions/builtin/github/helpers.ts [hunk 1]
```diff
@@ -349,10 +350,8 @@ export async function initRepoPushAndProtect(
       : config.getOptionalString('scaffolder.defaultAuthor.email'),
   };
 
-  const commitMessage = gitCommitMessage
-    ? gitCommitMessage
-    : config.getOptionalString('scaffolder.defaultCommitMessage');
-
+  const commitMessage =
+    getGitCommitMessage(gitCommitMessage, config) || 'initial commit';
```
[the remark this answers is not in the document]
Comment (npiyush97) [reply plugins/scaffolder-backend/src/scaffolder/actions/builtin/github/helpers.ts:354]: have a look

**Comment by npiyush97** [comment]
@benjdlambert merge ?

**Comment by benjdlambert** [comment]
@npiyush97 I just want to maybe hold off on this until we've got a review on #21752 as it's going to be easier to fix this merge conflict rather than the other big PR 😅

**Comment by awanlin** [comment]
Hi @npiyush97, now that https://github.com/backstage/backstage/pull/21752 has been merged in you'll need to rebase but also need to move where these changes are being made

**Pull request closed without merging**